src/_pytest/terminal.py

```diff
@@ -956,7 +956,28 @@ def show_skipped(lines):
 
 def _get_pos(config, rep):
     nodeid = config.cwd_relative_nodeid(rep.nodeid)
-    return nodeid
+    path, _, testname = nodeid.partition("::")
+    if not testname:
+        return nodeid
+
+    # Append location (line number).
+    # This uses the first traceback entry for the location in the test itself
+    # (rather than reprcrash, which might be less relevant for going to
+    # directly, e.g. pexpect failures in pytest itself).
+    try:
+        testloc = rep.longrepr.reprtraceback.reprentries[0].reprfileloc
+    except AttributeError:
+        return nodeid
+
+    if isinstance(testloc.path, six.string_types):
+        testloc_path = py.path.local(testloc.path)
+    else:
+        testloc_path = testloc.path
+    rel_testloc_path = config.invocation_dir.bestrelpath(testloc_path)
+    if rel_testloc_path == path:
+        return "%s (line %d)" % (nodeid, testloc.lineno)
+
+    return "%s (%s:%d)" % (nodeid, rel_testloc_path, testloc.lineno)
```

Comment:
Does this make sense like this?
Is there a better API (to get to the entry in the test)?